overlord/snapstate/snapstate.go

```diff
@@ -1761,8 +1763,16 @@ func ConfigDefaults(st *state.State, snapName string) (map[string]interface{}, e
 		return nil, err
 	}
 
+	core, err := CoreInfo(st)
+	if err != nil {
+		return nil, err
+	}
+	isCoreDefaults := core.Name() == snapName
+
 	si := snapst.CurrentSideInfo()
-	if si.SnapID == "" {
+	// core snaps can be addressed even without a snap-id via the special
+	// "system" value in the config
```

Comment:
I had to chase how this works,  probably worth adding

// first-boot always configures the core snap with UseConfigDefaults

Basically we don't need to worry  about:

https://github.com/bboozzoo/snapd/blob/e1ebc92eb7355c1202ad42ed071c385aaa280d14/overlord/snapstate/snapstate.go#L275